### PR TITLE
Upgrade Bazel's embedded JDK to 21.

### PR DESCRIPTION
**Pull request by zhengwei143**

### CODE REVIEW

**Review by coeuvre — APPROVED** [review]
Thanks!

File: distdir_deps.bzl [hunk 1]
```diff
@@ -488,39 +489,39 @@ DIST_DEPS = {
...
-            "https://mirror.bazel.build/cdn.azul.com/zulu/bin/zulu20.30.11-ca-jdk20.0.1-win_x64.zip",
-            "https://cdn.azul.com/zulu/bin/zulu20.30.11-ca-jdk20.0.1-win_x64.zip",
+            "https://mirror.bazel.build/cdn.azul.com/zulu/bin/zulu21.28.85-ca-jdk21.0.0-win_x64.zip",
+            "https://cdn.azul.com/zulu/bin/zulu21.28.85-ca-jdk21.0.0-win_x64.zip",
         ],
         "used_in": [
         ],
     },
-    # JDK20 unavailable so use JDK19 instead for win aarch64.
+    # JDK21 unavailable so use JDK19 instead for win aarch64.
```
Comment (fmeum) [inline distdir_deps.bzl:524]: We should keep in mind that this means that using JDK 21 features such as virtual threads will fail at runtime with Win aarch64 builds of Bazel.

How about using Microsoft's build of the OpenJDK for this platform? https://learn.microsoft.com/en-us/java/openjdk/download#openjdk-21

CC @meteorcloudy

Comment (zhengwei143) [reply distdir_deps.bzl:524]: Yup! @coeuvre and I were considering this as well, as for linux s390x and ppc64le, we don't release any Bazel binaries for those so it should be fine to ignore those.

@meteorcloudy @meisterT any objections against using Microsoft's OpenJDK build?

Comment (meteorcloudy) [reply distdir_deps.bzl:524]: No

Comment (zhengwei143) [reply distdir_deps.bzl]: Done.

**Review by meteorcloudy — APPROVED** [review]

**Comment by zhengwei143** [comment]
Downstream pipeline seems fine, no new errors between this PR (https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/3340) vs HEAD (https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/3334#018ad466-3db3-43a8-a052-3177d8279b0a)

**Comment by EdSchouten** [comment]
@zhengwei143 With this change landed, is it also possible for us to start submitting code changes to Bazel that require JDK21? Or do we first need to wait one or more release cycles?

**Pull request closed without merging**